### DISCUSSION TITLE
validatesPresenceOf() with condition and relation table

### DISCUSSION THREAD

**Discussion by Renand** [wheels-dev/wheels#1461] [General]
Hello everyone, I am looking to make  mandatory a field in a customer account only if the account has a particular type. The type is in an association table.

`customertypes {customerid, typeid}
`

How can I use a condition attribution to do that ? If it's not possible, can a condition be the return of a function?

Thanks in advance.

**Comment by chapmandu** [comment]
Hello,

I'd recommend using a custom validator containing a some logic to check the other model rather than the condition argument. 

https://api.cfwheels.org/v2.5.0/model.validate.html

Hope this helps.
